src/dkg.ts

```diff
@@ -17,22 +18,15 @@ import { DkgCoordinatorAgent, DkgRitualState } from './agents/coordinator';
 import { ChecksumAddress } from './types';
 import { bytesEquals, fromHexString, objectEquals } from './utils';
 
-// TODO: Expose from @nucypher/nucypher-core
-export enum FerveoVariant {
-  Simple = 0,
-  Precomputed = 1,
-}
-
 export function getVariantClass(
   variant: FerveoVariant
 ): typeof DecryptionShareSimple | typeof DecryptionSharePrecomputed {
-  switch (variant) {
-    case FerveoVariant.Simple:
-      return DecryptionShareSimple;
-    case FerveoVariant.Precomputed:
-      return DecryptionSharePrecomputed;
-    default:
-      throw new Error(`Invalid FerveoVariant: ${variant}`);
+  if (variant.equals(FerveoVariant.simple)) {
```

Comment:
Just out of curiosity: why this have been changed from switch conditionals to If/ else if?

Reply:
Ok, got it. Thanks for the elaborate explanation, Piotr 🙌